model/fv_dynamics.F90

```diff
@@ -909,6 +913,29 @@ subroutine fv_dynamics(npx, npy, npz, nq_tot,  ng, bdt, consv_te, fill,
      endif
   endif
 
+
+!mzhang:F-A nwat=4
+  if( nwat == 4 ) then                                                                                                        
+        call neg_adj2(is, ie, js, je, ng, npz,        &                                                                       
+                      flagstruct%hydrostatic,         &                                                                       
+                      peln, delz,                     &                                                                       
+                      pt, delp, q(isd,jsd,1,sphum),   &                                                                       
+                                q(isd,jsd,1,liq_wat), &                                                                       
+                                q(isd,jsd,1,rainwat), &                                                                       
+                                q(isd,jsd,1,ice_wat), &                                                                       
+                                check_negative=flagstruct%check_negative)                                                     
+     if ( flagstruct%fv_debug ) then                                                                                          
+       call prt_mxm('T_dyn_a3',    pt, is, ie, js, je, ng, npz, 1., gridstruct%area_64, domain)                               
+       call prt_mxm('SPHUM_dyn',   q(isd,jsd,1,sphum  ), is, ie, js, je, ng, npz, 1.,gridstruct%area_64, domain)              
+       call prt_mxm('liq_wat_dyn', q(isd,jsd,1,liq_wat), is, ie, js, je, ng, npz, 1.,gridstruct%area_64, domain)              
+       call prt_mxm('rainwat_dyn', q(isd,jsd,1,rainwat), is, ie, js, je, ng, npz, 1.,gridstruct%area_64, domain)              
+       call prt_mxm('ice_wat_dyn', q(isd,jsd,1,ice_wat), is, ie, js, je, ng, npz, 1.,gridstruct%area_64, domain)              
+!       call prt_mxm('snowwat_dyn', q(isd,jsd,1,snowwat), is, ie, js, je, ng, npz, 1.,gridstruct%area_64, domain)              
```

Comment:
Ditto.

Reply:
I will delete this line since no snowat in HWRF

Reply:
Ok